source/utilities.tex

```diff
@@ -13475,13 +13475,37 @@
     size_t formatted_size(const locale& loc, @\exposid{wformat-string}@<Args...> fmt, Args&&... args);
 
   // \ref{format.formatter}, formatter
+  // \ref{format.formatter.spec}, formatter specializations
   template<class T, class charT = char> struct formatter;
 
+  // \ref{format.range.formatter}, class template \tcode{range_formatter}
+  template<class T, class charT = char>
+    requires @\libconcept{same_as}@<remove_cvref_t<T>, T> && @\libconcept{formattable}@<T, charT>
+  class range_formatter;
+
+  template<ranges::@\libconcept{input_range}@ R, class charT>
+    requires (!@\libconcept{same_as}@<remove_cvref_t<ranges::range_reference_t<R>>, R>) &&
+             @\libconcept{formattable}@<ranges::range_reference_t<R>, charT>
+  struct formatter<R, charT>;
```

Comment:
Would it makes sense to swap the order of these two declarations? We have the heading "formatter specializations" then the `formatter` primary template, then the primary template of something else, then finally we get a formatter specialization as promise by the heading! It might make more sense to have the formatter primary template, then the specialization, then the `range_formatter` primary template.

That's not what the [paper](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2286r8.html#formatting-for-ranges) says, but I think it makes more sense. @brevzin ?

Reply:
Works for me. As long as it's all there in some sensible order, I'm happy.

Reply:
done